From a7c6168dfea437e6bceb429c5fa1c068c2d68695 Mon Sep 17 00:00:00 2001
From: Zhuo Peng <1835738+brills@users.noreply.github.com>
Date: Mon, 26 Oct 2020 16:35:44 -0700
Subject: [PATCH 1/2] patch for MSVC

---
 cpp/src/arrow/type_fwd.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git cpp/src/arrow/type_fwd.h cpp/src/arrow/type_fwd.h
index fc25b2723..783bf3d07 100644
--- cpp/src/arrow/type_fwd.h
+++ cpp/src/arrow/type_fwd.h
@@ -25,6 +25,13 @@
 #include "arrow/util/macros.h"
 #include "arrow/util/visibility.h"
 
+// Work around C2528 error
+#ifdef _MSC_VER
+#if _MSC_VER >= 1900
+#undef timezone
+#endif
+#endif
+
 namespace arrow {
 
 template <typename T>
-- 
2.29.0.rc1.297.gfa9743e501-goog


From a458c591b237c0661e64971e8f4cce88e9de1886 Mon Sep 17 00:00:00 2001
From: Zhuo Peng <1835738+brills@users.noreply.github.com>
Date: Mon, 26 Oct 2020 18:00:36 -0700
Subject: [PATCH 2/2] uint128

---
 cpp/src/arrow/util/int128_internal.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git cpp/src/arrow/util/int128_internal.h cpp/src/arrow/util/int128_internal.h
index 1d7c89806..7146f10ec 100644
--- cpp/src/arrow/util/int128_internal.h
+++ cpp/src/arrow/util/int128_internal.h
@@ -23,7 +23,7 @@
 #if defined(__SIZEOF_INT128__) && !defined(ARROW_UBSAN)
 #define ARROW_USE_NATIVE_INT128
 #else
-#include <boost/multiprecision/cpp_int.hpp>
+#include "absl/numeric/int128.h"
 #endif
 
 namespace arrow {
@@ -40,8 +40,8 @@ namespace internal {
 using int128_t = __int128_t;
 using uint128_t = __uint128_t;
 #else
-using boost::multiprecision::int128_t;
-using boost::multiprecision::uint128_t;
+using int128_t = absl::int128;
+using uint128_t = absl::uint128;
 #endif
 
 }  // namespace internal
-- 
2.29.0.rc1.297.gfa9743e501-goog

